src/pages/InputDemo/InputDemo.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useState } from 'react';
+import * as yup from 'yup';
+import {
+  TextField, SelectField, RadioGroup, Button,
+} from '../../components';
+import {
+  selectOptions, radioOptionsCricket, radioOptionsFootball,
+} from '../../configs/Constants';
+
+const InputDemo = () => {
+  const schema = yup.object().shape({
+    name: yup.string().required('name is required').min(3, 'should have more then 3 characters'),
+    sport: yup.string().required('sport is required'),
+    cricket: yup.string().when('sport', { is: 'cricket', then: yup.string().required('required') }),
+    football: yup.string().when('sport', { is: 'football', then: yup.string().required('required') }),
+  });
+
+  const [state, setstate] = useState({
+    name: '', sport: '', cricket: '', football: '',
+  });
+
+  const [onBlur, setBlur] = useState({
+    name: false, sport: false, cricket: false, football: false,
+  });
+
+  const handleBlur = (label) => {
+    setBlur({ ...onBlur, [label]: true });
+  };
+
+  const isTouched = () => {
+    if (onBlur.name || onBlur.sport || onBlur.cricket || onBlur.football) {
+      return true;
+    }
+    return false;
+  };
+
+  const hasErrors = () => {
+    try {
+      schema.validateSync(state);
```

Comment:
use schema.validate method instead of validateSync as there are some limitations to validateSync